test/gpu/native/basics/bug23658.chpl

```diff
@@ -0,0 +1,24 @@
+// this code is from https://github.com/chapel-lang/chapel/issues/23658 verbatim
+// the bug reported will be resolved via
+// TBD
+use GpuDiagnostics;
+
+config const gpuDiags = true;
+const sz = 10;
+
+if gpuDiags then startGpuDiagnostics();
+
+on here.gpus[0] {
+  const s = sz;
+  var Arr: [0..<s] int;
+  foreach i in 1..0 do {
+    Arr[i] = i;
+  }
+  writeln(Arr);
+}
+
+if gpuDiags {
+  stopGpuDiagnostics();
+  assertGpuDiags(kernel_launch=1, host_to_device=0, device_to_host=sz,
```

Comment:
This 1 launch is coming from array initialization. I expect that you will _not_ get any launches with `unified_memory`. You can drop `kernel_launch` argument, and use `kernel_launch_aod` and `kernel_launch_um` args separately to set different expected values for those cases.

(This makes me think we should stop doing separate things and have `unified_memory` do GPU-based initialization)